From b7335df2dc1b4c6c69ac6b95910d09a7dd5295a7 Mon Sep 17 00:00:00 2001
From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Wed, 29 Jun 2016 10:38:52 -0400
Subject: [PATCH] rtl8xxxu: Add rtl8188e_disabled_to_emu()

This sequence is found in the vendor driver, but never actually
called. It's unclear if we need it.

Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
---
 drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_8188e.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

--- a/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_8188e.c
+++ b/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_8188e.c
@@ -92,6 +92,15 @@ static int rtl8188eu_load_firmware(struc
 	return ret;
 }
 
+static void rtl8188e_disabled_to_emu(struct rtl8xxxu_priv *priv)
+{
+	u16 val16;
+
+	val16 = rtl8xxxu_read16(priv, REG_APS_FSMCO);
+	val16 &= ~(APS_FSMCO_PFM_WOWL | APS_FSMCO_ENABLE_POWERDOWN);
+	rtl8xxxu_write16(priv, REG_APS_FSMCO, val16);
+}
+
 static int rtl8188e_emu_to_active(struct rtl8xxxu_priv *priv)
 {
 	u8 val8;
@@ -165,6 +174,8 @@ static int rtl8188eu_power_on(struct rtl
 {
 	int ret;
 
+	rtl8188e_disabled_to_emu(priv);
+
 	ret = rtl8188e_emu_to_active(priv);
 	if (ret)
 		goto exit;
